scripts/emulate_app.sh

```diff
@@ -249,46 +243,53 @@ else
-        fi
-        i=$((i+1))
-        sleep 1
-    done
+    if "${boot_app}"; then
+        echo -e "\nRunning yarn install..."
+        yarn install
+
+        echo -e "\nBuilding APKs..."
+        ./gradlew assembleDebug -Psplitapk=true
 
-    # Store emulator info and start logging
-    adb -s emulator-5554 shell getprop &> "${test_report_dir}/getprop.txt"
-    adb -s emulator-5554 shell cat /proc/meminfo &> "${test_report_dir}/meminfo.txt"
-    adb -s emulator-5554 shell cat /proc/cpuinfo &> "${test_report_dir}/cpuinfo.txt"
-    adb -s emulator-5554 shell logcat -v threadtime -b main &> "${test_report_dir}/logcat.txt" &
+        echo -e "\nInstalling APKs..."
+        i=0
+        step_complete=false
+        until [[ $step_complete == true ]]; do
+            if adb -s emulator-5554 install -r -t -d --abi "${abi}" \
+                    "app/build/outputs/apk/debug/app-${abi}-debug.apk" &> "${test_report_dir}/apk_installation.txt"; then
+                step_complete=true
+                echo "Successfully installed APKs"
+            fi              
+            if [[ $i -ge 100 ]]; then
+                echo "Error: Failed to install APKs" >&2
+                echo "For more information see 'android/${test_report_dir}/apk_installation.txt'" >&2
+                exit 1
+            fi
+            i=$((i+1))
+            sleep 1
+        done
+
+        # Store emulator info and start logging
+        adb -s emulator-5554 shell getprop &> "${test_report_dir}/getprop.txt"
+        adb -s emulator-5554 shell cat /proc/meminfo &> "${test_report_dir}/meminfo.txt"
+        adb -s emulator-5554 shell cat /proc/cpuinfo &> "${test_report_dir}/cpuinfo.txt"
+        adb -s emulator-5554 shell logcat -v threadtime -b main &> "${test_report_dir}/logcat.txt" &
     
-    # Start react-native
-    if killall node; then
-        echo -e "\nAll node processes killed."
-        echo -e "\nRestarting react native..."
-    fi
-    nohup yarn react-native start > "${test_report_dir}/react_native.out" &> /dev/null &
+        # Start react-native
+        if killall node; then
+            echo -e "\nAll node processes killed."
+            echo -e "\nRestarting react native..."
+        fi
+        nohup yarn react-native start > "${test_report_dir}/react_native.out" &> /dev/null &
         
-    echo -e "\nLaunching App..."
-    nohup adb shell am start -n "org.ZingoLabs.Zingo/org.ZingoLabs.Zingo.MainActivity" -a android.intent.action.MAIN \
-        -c android.intent.category.LAUNCHER > "${test_report_dir}/launch_app.out" &> /dev/null &
+        echo -e "\nLaunching App..."
+        nohup adb shell am start -n "org.ZingoLabs.Zingo/org.ZingoLabs.Zingo.MainActivity" -a android.intent.action.MAIN \
+            -c android.intent.category.LAUNCHER > "${test_report_dir}/launch_app.out" &> /dev/null &
 
-    echo -e "\nTest reports saved: android/${test_report_dir}"        
+        echo -e "\nTest reports saved: android/${test_report_dir}"        
```

Comment:
as there are things saved to test reports outside of this `boot_app` section, can we put this after we exit this if statement?

also, i just realised it was my mistake but we are using the same directory as integration tests where i copied the script. it would be a nice improvement to have a whole new directory for emulating manually and changing the comment here also:
https://github.com/zingolabs/zingo-mobile/pull/330/files/53aef2b69b6c1e1aae75a54e3725edb11a997cd2..5eb69bf6e343c99e65869837ff3b8d6ba816505b#diff-c978b23d882c1e0ea4a509c0d51c9294acee8ba79ec8a92a0650696ee17b18e7R233